src/neo/IO/Caching/HashSetCache.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Neo.IO.Caching
+{
+    public class HashSetCache<T> : IReadOnlyCollection<T> where T : IEquatable<T>
+    {
+        private readonly int bucketCapacity;
+        private readonly int bucketCount;
+        private readonly List<HashSet<T>> sets = new List<HashSet<T>>();
```

Comment:
Maybe it is better to use a `LinkedList<T>`.

Reply:
Good idea, I'll try it.

Reply:
After test LinkedList have almost the same performance this List, but using LinkedList here is more appropriate.